aws/certificate/main.tf

```diff
@@ -0,0 +1,44 @@
+locals {
+  domain_names_parts = {
+    for domain in var.domain_names: domain => split(".", domain)
+  }
+
+  zone_names = {
+    for domain, parts in local.domain_names_parts:
+      domain => join(".", length(parts) > 2 ? slice(parts, 1, length(parts)) : parts)
+  }
+}
+
+data "aws_route53_zone" "validation_zones" {
+  for_each = toset(values(local.zone_names))
+
+  name = each.value
+}
```

Comment:
Impressive. I honestly wouldnt have gone this far, but it is very nice

Reply:
Thanks, I also think it turned out okay! 😄